lib/libzfs/libzfs_mount.c

```diff
@@ -972,9 +972,21 @@ zfs_unshareall(zfs_handle_t *zhp)
 }
 
 int
-zfs_unshareall_bypath(zfs_handle_t *zhp, const char *mountpoint)
+zfs_unshareall_bypath(zfs_handle_t *zhp, const char *mountpoint,
+    const char *proto)
 {
-	return (zfs_unshare_proto(zhp, mountpoint, share_all_proto));
+	zfs_share_proto_t *curr_proto = NULL;
+
+	if (proto == NULL)
+		curr_proto = share_all_proto;
```

Comment:
You should be able to simplify this by initializing `curr_proto = share_all_proto` and dropping this first conditional.